content/docs/latest/installing/vms/hyper-v.md

```diff
@@ -0,0 +1,119 @@
+
+For more information on customization that can be done with cloud-config, head on over to the [cloud-config guide](https://github.com/flatcar/coreos-cloudinit/blob/master/Documentation/cloud-config.md).
+
+You need a config-drive to configure at least one SSH key to access the virtual machine. If you are in hurry, you can create a basic config-drive with following steps:
+
+```powershell
+#ps1
+# mkisofs can be downloaded from https://github.com/cloudbase/cloudbase-init-test-resources/tree/master/bin
+# create an Openstack config drive folder structure
+mkdir config-drive-metadata/openstack/latest
+echo '{"hostname": "my_flatcar_01.local", "name": "my_flatcar_01", "public_keys": {"userkey": "INSERT_HERE_PUBLIC_SSH_KEY"}' > config-drive-metadata/openstack/latest/meta_data.json
+# create the config drive
+& "mkisofs.exe" -o "config-drive.iso" -ignore-error -ldots -allow-lowercase -allow-multidot -l -publisher "cbsl" -quiet -J -r -V "config-2" "config-drive-metadata"
+```
+
+An ISO file named `config-drive.iso` will be created that will configure a virtual machine to accept your SSH key and set its name to my_flatcar_01.
+
+## Deploying a new virtual machine on Hyper-V using config drive
+
+```powershell
+$vmName = "my_flatcar_01"
+$vmDisk = "flatcar_production_hyperv_vhdx_image.vhdx"
+
+New-VM -Name $vmName -MemoryStartupBytes 2GB `
+    -BootDevice VHD -SwitchName "Default Switch" -VHDPath $vmDisk -Generation 2
+Add-VMDvdDrive -VMName $vmName -Path "config-drive.iso"
+Set-VMFirmware -EnableSecureBoot "Off" -VMName $vmName
+
+Start-VM -Name $vmName
+```
+
+## Deploying a new virtual machine on Hyper-V using Ignition for TPM LUKS2 root partition encryption
+
+```powershell
+$vmName = "my_flatcar_01"
+$vmDisk = "flatcar_production_hyperv_vhdx_image.vhdx"
+
+New-VM -Name $vmName -MemoryStartupBytes 2GB `
+    -BootDevice VHD -SwitchName "Default Switch" -VHDPath $vmDisk -Generation 2
+Set-VMFirmware -EnableSecureBoot "Off" -VMName $vmName
+
+# download the tool kvpctl to set the Ignition metadata from
+# https://github.com/containers/libhvee/releases
+$ignitionMetadata = @'
+{
+  "ignition": {
+    "version": "3.4.0"
+  },
+  "storage": {
+    "filesystems": [
+      {
+        "device": "/dev/mapper/rootencrypted",
+        "format": "ext4",
+        "label": "ROOT"
+      }
+    ],
+    "luks": [
+      {
+        "device": "/dev/disk/by-partlabel/ROOT",
+        "name": "rootencrypted",
+        "wipeVolume": true
+      }
+    ]
+  },
+  "systemd": {
+    "units": [
+      {
+        "contents": "[Unit]\nConditionFirstBoot=true\nOnFailure=emergency.target\nOnFailureJobMode=isolate\n[Service]\nType=oneshot\nRemainAfterExit=yes\nExecStart=systemd-cryptenroll --tpm2-device=auto --unlock-key-file=/etc/luks/rootencrypted --wipe-slot=0 /dev/disk/by-partlabel/ROOT\nExecStart=rm /etc/luks/rootencrypted\n[Install]\nWantedBy=multi-user.target\n",
+        "enabled": true,
```

Comment:
Sure, will do. I'll add a base autologin ignition example and then ref the TPM example as a more complex scenario.

Reply:
or do you prefer to adapt the hyperv oem config?

Reply:
I can create something similar to https://github.com/flatcar/scripts/tree/main/sdk_container/src/third_party/coreos-overlay/coreos-base/oem-azure, where I can add the required guest services for communication (the so-called integration services). https://wiki.gentoo.org/wiki/Hyper-V#Linux_guest_support See: https://github.com/LIS/lis-pipeline/blob/master/scripts/package_building/deps-lis/rpm/lis-daemon.spec#L2

But I will update this document in another PR after the feature is implemented in flatcar/scripts.

Reply:
> Instead of having plain Ignition here we can link to the config section of the new page introduced in #317

Will add the link once https://github.com/flatcar/flatcar-website/pull/317 gets merged

Reply:
A link to the Windows binaries would be good here, too: https://github.com/coreos/butane/releases/tag/v0.20.0

Reply:
@pothos I added the link to the butane releases and also the Ignition metadata is now in butane format + added too the transformation command.